File: articles/designing-environments-that-support-grit/fortran/environment_design_index.f90
program environment_design_index
  implicit none
  real :: index

  index = 0.13 * 4.2 + 0.13 * 4.1 + 0.13 * 4.3 + 0.12 * 4.1 + &
          0.10 * 3.9 + 0.13 * 4.0 + 0.10 * 3.8 + 0.11 * 4.1 + &
          0.10 * 4.2 + 0.05 * 3.9

  print *, "Synthetic environment design index:", index
  print *, "Professional caution: synthetic demonstration only."
end program environment_design_index

File: articles/designing-environments-that-support-grit/rust/environment_design_index.rs
fn environment_design_index(
    autonomy: f64,
    competence: f64,
    feedback: f64,
    belonging: f64,
    mentoring: f64,
    recovery: f64,
    resources: f64,
    fairness: f64,
    safety: f64,
    adaptive_quitting: f64,
) -> f64 {
    0.13 * autonomy
        + 0.13 * competence
        + 0.13 * feedback
        + 0.12 * belonging
        + 0.10 * mentoring
        + 0.13 * recovery
        + 0.10 * resources
        + 0.11 * fairness
        + 0.10 * safety
        + 0.05 * adaptive_quitting
}

fn main() {
    let index = environment_design_index(4.2, 4.1, 4.3, 4.1, 3.9, 4.0, 3.8, 4.1, 4.2, 3.9);
    println!("Synthetic environment design index: {:.3}", index);
    println!("Professional caution: synthetic demonstration only.");
}
